packages/api/models/pdrs.js

```diff
@@ -100,24 +100,44 @@ class Pdr extends Manager {
     };
 
     record.duration = (record.timestamp - record.createdAt) / 1000;
+    this.constructor.recordIsValid(record, this.schema);
     return record;
   }
 
   /**
-   * Create a new PDR record from incoming SNS messages
+   * Try to update a PDR record from a cloudwatch event.
+   * If the record already exists, only update if the execution is different (re-run case).
    *
-   * @param {Object} payload - SNS message containing the output of a Cumulus Step Function
-   * @returns {Promise<Object>} a PDR record
+   * @param {Object} cumulusMessage - cumulus message object
    */
-  createPdrFromSns(payload) {
-    const pdrObj = get(payload, 'payload.pdr', get(payload, 'meta.pdr'));
-    const pdrName = get(pdrObj, 'name');
-
-    if (!pdrName) return Promise.resolve();
-
-    const pdrRecord = Pdr.generatePdrRecord(payload);
+  async storePdrsFromCumulusMessage(cumulusMessage) {
```

Comment:
So the relationship of a cumulus message to a PDR is 1:1, correct? Not sure if we want to change the name here to `storePdrFromCumulusMessage` 🐜

Reply:
Yeah, I guess so. Makes me think about whether to support creating a record for each PR in the output from `DiscoverPdrs`... 🤔

Reply:
Wouldn't each PDR in `DiscoverPdrs` trigger a separate `ParsePdr` workflow and the processing of those workflows would create the individual PDR records?

Reply:
Yeah.. I was thinking about what if `QueuePdrs` fails, but I guess we wouldn't want a bunch of `running` records with the wrong execution and no information on stats.